v2/sourcedb-to-spanner/src/main/java/com/google/cloud/teleport/v2/source/reader/io/jdbc/rowmapper/provider/PostgreSQLJdbcValueMappings.java

```diff
@@ -19,34 +19,216 @@
+          .put("ENUM", Pair.of(ResultSet::getString, valuePassThrough))
+          .put("FLOAT4", Pair.of(ResultSet::getFloat, valuePassThrough))
+          .put("FLOAT8", Pair.of(ResultSet::getDouble, valuePassThrough))
+          .put("INET", Pair.of(ResultSet::getString, valuePassThrough))
+          .put("INT", Pair.of(ResultSet::getInt, valuePassThrough))
+          .put("INTEGER", Pair.of(ResultSet::getInt, valuePassThrough))
+          // TODO(thiagotnunes): INTERVAL
+          .put("INT2", Pair.of(ResultSet::getInt, valuePassThrough))
+          .put("INT4", Pair.of(ResultSet::getInt, valuePassThrough))
           .put("INT8", Pair.of(ResultSet::getLong, valuePassThrough))
+          .put("JSON", Pair.of(ResultSet::getString, valuePassThrough))
+          .put("JSONB", Pair.of(ResultSet::getString, valuePassThrough))
+          .put("LINE", Pair.of(ResultSet::getString, valuePassThrough))
+          .put("LSEG", Pair.of(ResultSet::getString, valuePassThrough))
+          .put("MACADDR", Pair.of(ResultSet::getString, valuePassThrough))
+          .put("MACADDR8", Pair.of(ResultSet::getString, valuePassThrough))
+          .put("MONEY", Pair.of(ResultSet::getDouble, valuePassThrough))
+          .put("NUMERIC", Pair.of(ResultSet::getObject, numericToAvro))
+          .put("OID", Pair.of(ResultSet::getLong, valuePassThrough))
+          .put("PATH", Pair.of(ResultSet::getString, valuePassThrough))
+          .put("PG_LSN", Pair.of(ResultSet::getString, valuePassThrough))
+          .put("PG_SNAPSHOT", Pair.of(ResultSet::getString, valuePassThrough))
+          .put("POINT", Pair.of(ResultSet::getString, valuePassThrough))
+          .put("POLYGON", Pair.of(ResultSet::getString, valuePassThrough))
+          .put("REAL", Pair.of(ResultSet::getFloat, valuePassThrough))
+          .put("SERIAL", Pair.of(ResultSet::getInt, valuePassThrough))
+          .put("SERIAL2", Pair.of(ResultSet::getInt, valuePassThrough))
+          .put("SERIAL4", Pair.of(ResultSet::getInt, valuePassThrough))
+          .put("SERIAL8", Pair.of(ResultSet::getLong, valuePassThrough))
+          .put("SMALLINT", Pair.of(ResultSet::getInt, valuePassThrough))
+          .put("SMALLSERIAL", Pair.of(ResultSet::getInt, valuePassThrough))
           .put("TEXT", Pair.of(ResultSet::getString, valuePassThrough))
+          .put("TIME", Pair.of(timeExtractor, timeToAvro))
+          .put("TIMETZ", Pair.of(timetzExtractor, timetzToAvro))
+          .put("TIMESTAMP", Pair.of(timestampExtractor, timestampToAvro))
+          .put("TIMESTAMPTZ", Pair.of(timestamptzExtractor, timestamptzToAvro))
+          .put("TIME WITH TIME ZONE", Pair.of(timetzExtractor, timetzToAvro))
+          .put("TIME WITHOUT TIME ZONE", Pair.of(timeExtractor, timeToAvro))
+          .put("TIMESTAMP WITH TIME ZONE", Pair.of(timestamptzExtractor, timestamptzToAvro))
+          .put("TIMESTAMP WITHOUT TIME ZONE", Pair.of(timestampExtractor, timestampToAvro))
```

Comment:
[nit and can be taken up later], should we normalize `TIMESTAMP WITHOUT TIME ZONE` to `TIMESTAMP` and similarly for `TIME WITHOUT TIME ZONE` to `TIME` in schema discovery?

Reply:
Good point we can reduce the testing then.